### DISCUSSION TITLE
FragAttacks

### DISCUSSION THREAD

**Discussion by brenard** [arendst/Tasmota#12104] [Q&A] [answered]
Hello,

I would like to know is Tasmota is concerned [FragAttacks vulnerability](https://www.fragattacks.com/)? If so, is there any work in progress on these topics? 

Thank you

Benjamin

**Comment by sfromis** [comment, marked as answer]
Strictly speaking, this is not a Tasmota question, as Tasmota sits of top of [Arduino Core](https://github.com/esp8266/Arduino) handling the wifi activity. Any concerns or work in progress would be handled in that project. In case of security fixes there, the Tasmota project will obviously look into getting these into the binaries built and published.

**Reply by brenard** [reply]
OK, thanks. I will ask for it to arduino core project.

**Reply by sfromis** [reply]
https://github.com/esp8266/Arduino/issues/8042

**Reply by brenard** [reply]
Exactly, thanks !

**Reply by brenard** [reply]
Subject isn't treated by esp8366/Arduino project, it's seen to be handled by ESP SDK. I posted a question on Expressif official forum (currently waiting for moderation). I will post a link to this question when it's availabled online.

**Reply by sfromis** [reply]
"we are looking into the testcases of frag attacks to see if we are vulnerable."
https://github.com/espressif/esp-idf/issues/7019

**Comment by flinty32** [comment]
If it's Part off of Arduino Core, when go Tasmota to the newer Arduino Core?  newer as 3.0.0 ?

**Reply by sfromis** [reply]
That depends on what fixes are present and useful to Tasmota, and if the cost in additional space used is too high. Sometimes fixes are instead backported to the fork Tasmota use for regular compiles.

**Reply by flinty32** [reply]
Ok, Thanks.

**Reply by Jason2866** [reply]
If you have opened a issue in Arduino Esp8266 please do reference to this discussion.
I will track. If it is solved, i try to do a backport to Tasmota ESP8266 2.7.4.x.
If a backport is not possible, we will discuss further what we gonna do.
Leaving the current builded Tasmota core 2.7.4.9 would have a big impact in code size and reliability. We optimized the core over a long time to get the best performance AND reliability for Tasmota use. We will not move to a other core without a heavy reason.

**Comment by tomacs** [comment]
I really not an expert about Expressif, Arduino etc. and how everything is build.

But it seems that there will be a fix:

https://github.com/espressif/esp-idf/issues/7019#issuecomment-870345579

Does this means that it will be one day part of Tasmota?

**Reply by Jason2866** [reply]
For Arduino ESP8266 it looks like there will be no fix since it is based on NonOsSDK 2.2.x
https://www.espressif.com/sites/default/files/advisory_downloads/AR2021-003%20Security%20Advisory%20for%20WFA%20vulnerability.pdf
As soon the fix for IDF 3.3.x is available we will compile and test a Arduino ESP32 Framework with.
If Tasmota32 is working as expected, the Tasmota32 development build will be based on this "core"

**Reply by tomacs** [reply]
Now it seems to be more clear for me. Thx for fast reply and for the hint....

I hope there will be a fix in NonOsSDK 3 that can be backported. 

https://github.com/espressif/ESP8266_NONOS_SDK/issues/343

Otherwise i need to think about the use of ESP8266 based devices... :-(

**Reply by Jason2866** [reply]
No backport. NONOS-SDK is closed source and there are only precompiled libs.
Libs from different NONOS-SDKs are not interchangable.
If espressif does not change its mind to do for NONOS-SDK 2.2.x too it will be left open
Do not expect a Arduino Esp8266 core based on NONOS-SDK 3.x it would be a complete rewrite for an old chip... This will not happen. 
There is a soulition in sight :-)  Tasmota is already running on the ESP32-C3 The naming of the chip is misleading. It is a single core MCU. Imho the "new" ESP8266 for diy

**Reply by sfromis** [reply]
While the thought of being hacked is of course not pleasant, FragAttacks is a low risk vulnerability due to the need to be in proximity of the wifi network. I'm not losing any sleep over that risk. 

"Fortunately, the **design flaws are hard to abuse** because doing so requires user interaction or is only possible when using uncommon network settings."

**Comment by nagyrobi** [comment]
I'm thinking of [this](https://www.fragattacks.com/):
_How can the adversary construct unencrypted Wi-Fi frames so they are accepted by a vulnerable device? First, certain Wi-Fi devices accept any unencrypted frame even when connected to a protected Wi-Fi network. This means the attacker doesn't have to do anything special! Two of out of four tested home routers were affected by this vulnerability, several internet-of-things devices were affected, and some smartphones were affected._

Tasmota can execute simply commands by http. So thir iot example from the video could be ran against Tasmota very easily. A few minutes of sniffing in the air would offer just enough information to open a wifi-lock with this method. A laptop running in a car parking in the street is enough...

We should compile a best practice recommendation to the users, like disable http commands, use mqtts, etc.

**Reply by ascillato** [reply]
> ... best practice recommendation ...

Those are already in the docs: https://tasmota.github.io/docs/Securing-your-IoT-from-hacking/

**Reply by sfromis** [reply]
Did you read my quote **from the same page** about the flaws being hard to abuse? Have you any reason to assume that ESP8266 firmware is included in that "certain Wi-Fi devices". Don't dream up a risk without hard facts.

**Reply by nagyrobi** [reply]
_hard to abuse_ is a relative thing.

**Reply by sfromis** [reply]
When the discoverers of the attack describe it as "hard to abuse", it is from a pretty high level of technical knowledge.

**Reply by nagyrobi** [reply]
That was true when that doc was written originally.

**Reply by sfromis** [reply]
If you have actual knowledge of *anything* interesting there, don't just make very vague musings. Like an *actual* documented risk.
